docs/reference/query-dsl/semantic-query.asciidoc

```diff
@@ -0,0 +1,148 @@
+[[query-dsl-semantic-query]]
+=== Semantic query
+++++
+<titleabbrev>Semantic</titleabbrev>
+++++
+
+beta[]
+
+The `semantic` query type enables you to perform semantic search on data stored
+in a <<semantic-text,`semantic_text`>> field.
+
+
+[discrete]
+[[semantic-query-example]]
+==== Example request
+
+[source,console]
+------------------------------------------------------------
+GET my-index-000001/_search
+{
+  "query": {
+    "semantic": {
+      "field": "inference_field",
+      "query": "Best surfing places"
+    }
+  }
+}
+------------------------------------------------------------
+// TEST[skip:TBD]
+
+
+[discrete]
+[[semantic-query-params]]
+==== Top-level parameters for `semantic`
+
+field::
+(Required, string)
+The `semantic_text` field to perform the query on.
+
+query::
+(Required, string)
+The query text to be searched for on the field.
+
+
+Refer to <<semantic-search-semantic-text,this tutorial>> to learn more about
+semantic search using `semantic_text` and `semantic` query.
+
+[discrete]
+[[hybrid-search-semantic]]
+==== Hybrid search with the `semantic` query
+
+The `semantic` query can be used as a part of a hybrid search where `semantic`
+query is combined with lexical queries. For example, the query below finds
+documents with title matching "mountain lake", and combines them with doing the
+same semantic search on the field `title_semantic`, that is a `semantic_text`
+field. The combined documents are then scored, and the top 3 top scored
+documents are returned.
+
+[source,console]
+------------------------------------------------------------
+POST my-index/_search
+{
+  "size" : 3,
+  "query": {
+    "bool": {
+      "should": [
+        {
+          "match": {
+            "title": {
+              "query": "mountain lake",
+              "boost": 1
+            }
+          }
+        },
+        {
+          "semantic": {
+            "field": "title_semantic",
+            "query": "mountain lake",
+            "boost": 2
+          }
+        }
+      ]
+    }
+  }
+}
+------------------------------------------------------------
+// TEST[skip:TBD]
+
+
+[discrete]
+[[search-sparse-inference]]
+==== Search with `sparse_embedding` inference
```

Comment:
Do you think it would be a good idea to add a paragraph explaining about the limitations of semantic query, that will serve as the intro for using sparse and dense inference searches?

Reply:
++

Reply:
@carlosdelest  Yes, we can add a separate section called `Limitations` to this page. Can you point me to a source of info so I know the content of such a section?

Reply:
My opinion: We should add a section called "Customizing the `semantic` query" just before "Search with `sparse_embedding` inference". We can use this section to explain why we didn't add a bunch of params to the `semantic` query:

```
==== Customizing the `semantic` query

You may have noticed that the `semantic` query doesn't provide many top-level parameters for customization.
This is intentional; it is intended to be used as a stepping stone to simplify the setup & evaluation of semantic search.
Once you have a good understanding of how to best apply semantic search in your use case, we recommend graduating to using the `sparse_vector` or `knn` query, depending on the inference type.
These query types provide greater flexibility, allowing you to customize behavior for better relevance and/or performance.
```

@carlosdelest WDYT?

Reply:
We had an intro text at the [original doc](https://docs.google.com/document/d/1DYhE6tjw6Bdh1Ctr2-5S-FBW7ZhV5El-jia78i07bMc/edit#heading=h.en3pyq5aexc6) we used. I'm happy with a mix of that and Mike's paragraph in the example.

I trust your judgement here @szabosteve !

Reply:
I like the text in the original doc as well, and I think "Advanced search on semantic_text fields" is a better heading. Combine as you see fit @szabosteve!

Reply:
Sorry, my mistake. I left our `Advanced search on semantic_text fields` by accident. I've added the section to my latest commit, combined the text with Mike's suggestion, and made it as concise as possible.